src/words/tweet_tokenizer.jl

```diff
@@ -0,0 +1,271 @@
+                        is_base_16 = false
+                        break
+                    end
+                end
+                if is_base_16
+                    number = parse(Int, entity_body, base=16)
+                end
+            end
+
+            # Numeric character references in the 80-9F range are typically
+            # interpreted by browsers as representing the characters mapped
+            # to bytes 80-9F in the Windows-1252 encoding. For more info
+            # see: http://en.wikipedia.org/wiki/Character_encodings_in_HTML
+
+            if 0x80 <= number <= 0x9F
+                if !(number in [129 141 143 144 157])
+                    return decode([UInt8(number)], "WINDOWS-1252")
+                end
+            else
+                if Unicode.isassigned(number)
+                    return (Char(number))
+                end
+            end
+        end
+
+        if remove_illegal
+            return ""
+        else
+            return matched_text
+        end
+    end
+
+    entities_replaced_text = replace(input_text, HTML_ENTITIES_REGEX => convert_entity)
+end
+
+
+"""
+    tweet_tokenize(input::AbstractString) => tokens
+
+Twitter-aware tokenizer, designed to be flexible and
+easy to adapt to new domains and tasks.
+
+The basic logic is following:
+
+1. The regular expressions are made for WORD_REGEX (core tokenizer), HANG_REGEX
+   and EMOTICONS_REGEX.
+2  Replacing HTML entities, tweet handles, reducing length of repeated characters
+   and other features, make it suitable for tweets.
+3. The String is tokenized and returned.
+4. `preserve_case` By default is set to `true`. If it is set to `false`,
+   then the tokenizer will downcase everything except for emoticons.
+
+Example:
+
+```
+julia> tweet_tokenize("This is a cooool #dummysmiley: :-) :-P <3 and some arrows < > -> <--")
+
+16-element Array{SubString{String},1}:
+ "This"
+ "is"
+ "a"
+ "cooool"
+ "#dummysmiley"
+ ":"
+ ":-)"
+ ":-P"
+ "<3"
+ "and"
+ "some"
+ "arrows"
+ "<"
+ ">"
+ "->"
+ "<--"
+```
+"""
+function tweet_tokenize(source::AbstractString;
+                            strip_handle=false,
+                            reduce_len=false,
+                            preserve_case=true )
+
+    function reduce_lengthening(source::AbstractString)
+        replace(source, r"(.)\1{2,}" => s"\1\1\1")
+    end
+
+    function remove_handles(source::AbstractString)
+        replace(source, HANDLES_REGEX => " ")
+    end
+
+    # Fix HTML Character entities
+    source = replace_html_entities(source)
+    # Remove username handles
+    if strip_handle
+        source = remove_handles(source)
+    end
+    # Reduce Lengthening
+    if reduce_len
+        source = reduce_lengthening(source)
+    end
+    # Shorten some sequences of characters
+    safe_text = replace(source, r"""([^a-zA-Z0-9])\1{3,}""" => s"\1\1\1")
+    # Tokenize
+    tokens = collect((m.match for m = eachmatch(WORD_REGEX,
```

Comment:
`collect(a.b for a in 1:10)` is the same as `[a.b for a in 1:10]
the latter is perfered.

Also better style for  `for` loop to use `for m in eachmatch...` not `for m = eachmatch...`
The latter is a strange matlabism.